docs/src/manual.md

```diff
@@ -529,6 +529,19 @@ julia> suite
 As you might imagine, `BenchmarkGroup` supports a subset of Julia's `Associative` interface. A full list of
 these supported functions can be found [in the reference document](reference.md#benchmarkgrouptagsvector-datadict).
 
+One can also create a nested `BenchmarkGroup` simply by passing the keys, similar to how
+`mkdir -p` works:
+
+```julia
+suite2 = BenchmarkGroup()
+
+suite2["my"]["nested"]["benchmark"] = @benchmarkable sum(randn(32))
+```
+
+which will result in a hierarchical benchmark without us needing to create each `BenchmarkGroup` ourselves.
+(Note that keys are automatically created upon access, even if a key does not exist. Thus, if you wish
+to empty the unused keys, you can use `clear_empty!` to do so)
```

Comment:
`clear_empty!(suite)` or `clear_empty!(suite, key1, key2, ...)`?

Reply:
`clear_empty!` finds the keys for you (recursively). You would prefer it to be manually used?

Reply:
No the recursive behavior is needed, I was just suggesting you make it explicit :)

Reply:
Oh I see, sounds good!

Reply:
Done!